recipes-bsp/opensbi/opensbi_nezha-1.0.bb

```diff
@@ -0,0 +1,58 @@
+SUMMARY = "RISC-V Open Source Supervisor Binary Interface (OpenSBI)"
+DESCRIPTION = "OpenSBI aims to provide an open-source and extensible implementation of the RISC-V SBI specification for a platform specific firmware (M-mode) and a general purpose OS, hypervisor or bootloader (S-mode or HS-mode). OpenSBI implementation can be easily extended by RISC-V platform or System-on-Chip vendors to fit a particular hadware configuration."
+HOMEPAGE = "https://github.com/riscv/opensbi"
+LICENSE = "BSD-2-Clause"
+LIC_FILES_CHKSUM = "file://COPYING.BSD;md5=42dd9555eb177f35150cf9aa240b61e5"
+
+require recipes-bsp/opensbi/opensbi-payloads.inc
+
+inherit autotools-brokensep deploy
+
+# Mainline OpenSBI supports the C906 out of the box, but it needs a few tweaks
+# and a new reset driver for the sunxi watchdog.
+SRCREV = "c9024b561c01aa469bed3c2266d78e6ae76882ff"
+SRC_URI = " \
+    git://github.com/tekkamanninja/opensbi.git;branch=allwinner_d1 \
+"
```

Comment:
If it's only a few patches can we just apply them on top of mainline OpenSBI instead?

You can do it with something like this: https://github.com/riscv/meta-riscv/blob/master/recipes-kernel/linux/linux-mainline_5.4.bb#L12

Reply:
yeah it would be cool to use mainline + patches approach but not a big deal to begin with.

Reply:
I checked the https://github.com/tekkamanninja/opensbi at branch `allwinner_d1` and it looks like it has two commits needed to be applied at the top of the upstream OpenSBI v1.0. But `hardknott` version of Yocto has OpenSBI in version `v0.9` and doesn't have files which are patched:
```
ERROR: opensbi-0.9-r0 do_patch: Command Error: 'quilt --quiltrc /build/tmp/work/riscv64-nezha-linux/opensbi/0.9-r0/recipe-sysroot-native/etc/quiltrc push' exited with 0  Output:
stdout: Applying patch 0001-lib-utils-fdt-Require-match-data-to-be-const.patch
can't find file to patch at input line 20
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|From 65410b015fc16d1c47b1972f89a35f5c8dd10abe Mon Sep 17 00:00:00 2001
|From: Samuel Holland <samuel@sholland.org>
|Date: Tue, 19 Oct 2021 19:25:37 -0500
|Subject: [PATCH 1/2] lib: utils/fdt: Require match data to be const
|
|Match data stores hardware attributes which do not change at runtime, so
|it does not need to be mutable. Make it const.
|
|Signed-off-by: Samuel Holland <samuel@sholland.org>
|---
| include/sbi_utils/fdt/fdt_helper.h | 2 +-
| lib/utils/ipi/fdt_ipi_mswi.c       | 2 +-
| lib/utils/reset/fdt_reset_gpio.c   | 4 ++--
| 3 files changed, 4 insertions(+), 4 deletions(-)
|
|diff --git a/include/sbi_utils/fdt/fdt_helper.h b/include/sbi_utils/fdt/fdt_helper.h
|index 24fee7a339c3..fde00b72fc1d 100644
|--- a/include/sbi_utils/fdt/fdt_helper.h
|+++ b/include/sbi_utils/fdt/fdt_helper.h
--------------------------
No file to patch.  Skipping patch.
1 out of 1 hunk ignored
can't find file to patch at input line 33
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/lib/utils/ipi/fdt_ipi_mswi.c b/lib/utils/ipi/fdt_ipi_mswi.c
|index 1f0fda71c087..0176941a7d68 100644
|--- a/lib/utils/ipi/fdt_ipi_mswi.c
|+++ b/lib/utils/ipi/fdt_ipi_mswi.c
--------------------------
No file to patch.  Skipping patch.
1 out of 1 hunk ignored
can't find file to patch at input line 46
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/lib/utils/reset/fdt_reset_gpio.c b/lib/utils/reset/fdt_reset_gpio.c
|index d28b6f5ba958..bd2c62270645 100644
|--- a/lib/utils/reset/fdt_reset_gpio.c
|+++ b/lib/utils/reset/fdt_reset_gpio.c
--------------------------
No file to patch.  Skipping patch.
2 out of 2 hunks ignored
Patch 0001-lib-utils-fdt-Require-match-data-to-be-const.patch does not apply (enforce with -f)

stderr: 
ERROR: Logfile of failure stored in: /build/tmp/work/riscv64-nezha-linux/opensbi/0.9-r0/temp/log.do_patch.661
ERROR: Task (/work/poky/meta/recipes-bsp/opensbi/opensbi_0.9.bb:do_patch) failed with exit code '1'
```

Maybe I should try to rebase my changes to the top of the `master` where the `kirkstone` version of Yocto is used?

Reply:
yes master would be just fine. That will soon become kirkstone.

Reply:
@kraj @alistair23 I create new PR with rebased changes: #327

Reply:
@Cezarus27 lets close this PR then to avoid confusion.